src/solve-model.jl

```diff
@@ -183,24 +65,129 @@ function solve_model(
         return nothing
     end
 
-    dual_variables = @timeit to "compute_dual_variables" compute_dual_variables(model)
-
-    return Solution(
-        Dict(k => JuMP.value(v) for (k, v) in variables[:assets_investment].lookup),
-        Dict(k => JuMP.value(v) for (k, v) in variables[:assets_investment_energy].lookup),
-        Dict(k => JuMP.value(v) for (k, v) in variables[:flows_investment].lookup),
-        JuMP.value.(variables[:storage_level_rep_period].container),
-        JuMP.value.(variables[:storage_level_over_clustered_year].container),
-        JuMP.value.(model[:max_energy_over_clustered_year]),
-        JuMP.value.(model[:min_energy_over_clustered_year]),
-        JuMP.value.(variables[:flow].container),
-        JuMP.objective_value(model),
-        dual_variables,
+    return
+end
+
+"""
+    save_solution!(energy_problem; compute_duals = true)
+"""
+function save_solution!(energy_problem::EnergyProblem; compute_duals = true)
+    return save_solution!(
+        energy_problem.db_connection,
+        energy_problem.model,
+        energy_problem.variables,
+        energy_problem.constraints;
+        compute_duals,
     )
 end
 
 """
-    compute_dual_variables(model)
+    save_solution!(connection, model, variables, constraints; compute_duals = true)
+
+Saves the primal and dual variables values, in the following way:
+
+- For each variable in `variables`, get the solution value and save it in a
+column named `solution` in the corresponding dataset.
+- For each constraint in `constraints`, get the dual of each attached
+constraint listed in `constraint.constraint_names` and save it to the dictionary
+`constraint.duals` with the key given by the name.
+
+Notice that the duals are only computed if `compute_duals` is true.
+"""
+function save_solution!(connection, model, variables, constraints; compute_duals = true)
+    # Check if it's solved
+    status = JuMP.termination_status(model)
+    if status == JuMP.OPTIMIZE_NOT_CALLED
+        @error("The model was not solved yet, use one of the solve_model functions")
+        return
+    elseif status != JuMP.OPTIMAL
+        @error("The model was not solved to optimality, cannot get solution")
+        return
+    end
```

Comment:
Here the same, I propose simplifying like this:

```suggestion
    if !JuMP.is_solved_and_feasible(model; dual = compute_duals)
        @error("The model has a termination status: $JuMP.termination_status(model), with primal status $JuMP.primal_status(model), and dual status $JuMP.dual_status(model)")
        return
    end
```